src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt

```diff
@@ -0,0 +1,92 @@
+package uk.gov.justice.digital.hmpps.personrecord.service
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Isolation
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonAddressEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonAliasEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonContactEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.Person
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+
+@Service
+class PersonService(
+  private val personRepository: PersonRepository,
+  private val telemetryService: TelemetryService,
+) {
+
+  @Transactional(isolation = Isolation.SERIALIZABLE)
+  fun processPerson(person: Person, callback: () -> PersonEntity?) {
+    val existingPersonEntity: PersonEntity? = callback()
+    handlePerson(person, existingPersonEntity)
```

Comment:
bit of a lazy method now, we could inline `handlePerson`?

Reply:
In fact the method naming in here is a bit confusing at first glance - difficult to say without reading the code what these three are doing for example. We can look at this later of course

`handlePersonUpdate`
`updateExistingPersonEntity`
`updateAndSavePersonEntity`